src/application.rs

```diff
@@ -137,20 +174,85 @@ impl Application {
                 self.model.borrow_mut().update_app_list(app_list);
             }
 
+            IpcMessage::ZedAgentStatus(status) => {
+                debug!("Got ZedAgentStatus");
+                self.model.borrow_mut().update_zed_agent_status(status);
+            }
+
             _ => {
                 warn!("Unhandled IPC message: {:?}", msg);
             }
         }
     }
 
-    pub fn send_dpc(&self) {
-        if let Some(current_dpc) = self.raw_model.get_current_dpc() {
-            let mut dpc = current_dpc.clone();
-
-            dpc.key = "manual".to_string();
-            dpc.time_priority = chrono::Utc::now();
-
-            self.send_ipc_message(IpcMessage::new_request(Request::SetDPC(dpc)));
+    pub fn send_dpc(&mut self, old: InterfaceState, new: InterfaceState) {
+        let current_dpc = self.model.borrow().get_current_dpc().cloned();
+        if let Some(current_dpc) = current_dpc {
+            info!("send_dpc: Sending DPC for iface {}", &new.iface_name);
+            let dpc = current_dpc.clone();
```

Comment:
you clone current_dpc twice here, once 

```rust
let current_dpc = self.model.borrow().get_current_dpc().cloned();
```
once here, doesn't make sense to clone twice, maybe even to clone at all actually